packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -0,0 +1,442 @@
+  );
+  // appPackage
+  await fs.copy(path.join(projectPath, AppPackageFolderName), templateAppPackage);
+  await fs.rename(
+    path.join(templateAppPackage, "manifest.source.json"),
+    path.join(templateAppPackage, "manifest.template.json")
+  );
+
+  await moveIconsToResourceFolder(templateAppPackage);
+  if (hasProvision) {
+    const devProfile = path.join(fxPublishProfile, "profile.dev.json");
+    const devUserData = path.join(fxPublishProfile, "dev.userdata");
+    await fs.copy(path.join(fx, "new.env.default.json"), devProfile);
+    await fs.copy(path.join(fx, "default.userdata"), devUserData);
+    await removeExpiredFields(devProfile, devUserData);
+    await ensureActiveEnv(projectSettings);
+  }
+}
+async function moveIconsToResourceFolder(templateAppPackage: string): Promise<void> {
+  // see AppStudioPluginImpl.buildTeamsAppPackage()
+  const manifest: TeamsAppManifest = await readJson(
+    path.join(templateAppPackage, "manifest.template.json")
+  );
+  const hasColorIcon = manifest.icons.color && !manifest.icons.color.startsWith("https://");
+  const hasOutlineIcon = manifest.icons.outline && !manifest.icons.outline.startsWith("https://");
+  if (!hasColorIcon || !hasOutlineIcon) {
+    return;
+  }
+  // move to resources
+  const resource = path.join(templateAppPackage, "resources");
+  await fs.ensureDir(resource);
+  await fs.move(
+    path.join(templateAppPackage, manifest.icons.color),
+    path.join(resource, manifest.icons.color)
+  );
+
+  await fs.move(
+    path.join(templateAppPackage, manifest.icons.outline),
+    path.join(resource, manifest.icons.outline)
+  );
+  // update icons
+  manifest.icons.color = `resources/${manifest.icons.color}`;
+  manifest.icons.outline = `resources/${manifest.icons.outline}`;
+  await fs.writeFile(
+    path.join(templateAppPackage, "manifest.template.json"),
+    JSON.stringify(manifest, null, 4)
+  );
+}
+
+async function removeExpiredFields(devProfile: string, devUserData: string): Promise<void> {
+  // remove fx-resource-local-debug.trustDevCert, solution.programmingLanguage, fx-resource-function.defaultFunctionName
+  const profileData = await readJson(devProfile);
+  if (profileData[PluginNames.LDEBUG]) {
+    delete profileData[PluginNames.LDEBUG];
+  }
+  if (profileData[PluginNames.SOLUTION] && profileData[PluginNames.SOLUTION][programmingLanguage]) {
+    delete profileData[PluginNames.SOLUTION][programmingLanguage];
+  }
+  if (profileData[PluginNames.FUNC] && profileData[PluginNames.FUNC][defaultFunctionName]) {
+    delete profileData[PluginNames.FUNC][defaultFunctionName];
+  }
+  await fs.writeFile(devProfile, JSON.stringify(profileData, null, 4), { encoding: "UTF-8" });
+  const trustDevCertKey = `${PluginNames.LDEBUG}.trustDevCert`;
+  const secrets: Record<string, string> = deserializeDict(await fs.readFile(devUserData, "UTF-8"));
+
+  if (secrets[trustDevCertKey]) {
+    delete secrets[trustDevCertKey];
```

Comment:
We'd better also clean up other local debug items in `default.userdata`, e.g.:

```txt
fx-resource-local-debug.trustDevCert
fx-resource-local-debug.skipNgrok
fx-resource-local-debug.localBotEndpoint
fx-resource-local-debug.localAuthEndpoint
fx-resource-local-debug.localTabEndpoint
fx-resource-local-debug.localTabDomain
fx-resource-local-debug.localFunctionEndpoint
fx-resource-local-debug.localBotDomain
```

Reply:
thanks, will fix.